mcp - Refactor doc search to return human readable results. Simplify search and doc indexing

devinea opened the pull request:
mcp - Refactor doc search to return human readable results. Simplify search and doc indexing
 - doc search - return more Human readable responses instead of JSON etc..
 - Remove simple indexer based on the file store and rely only on embeddings.
 - Increase chunks size for better results
 - reduce embeddings package tgz size.

unseen1980 commented:
@devinea I added this https://github.com/SAP/open-ux-tools/pull/3680/commits/2b8f81f1424c931b594d74ae30aca9989cb7e5b1

It appears now like this
<img width="801" height="689" alt="Screenshot 2025-09-29 at 13 46 01" src="https://github.com/user-attachments/assets/f18b02ea-4237-46be-b02c-fa0a9936786e" />

devinea commented:
> @devinea I added this [2b8f81f](https://github.com/SAP/open-ux-tools/commit/2b8f81f1424c931b594d74ae30aca9989cb7e5b1)

@unseen1980 as discussed I reverted the change. Thanks for looking into it.

heimwege reviewed (commented):
See comments

devinea commented on packages/fiori-docs-embeddings/src/scripts/build-docs.ts:
```diff
@@ -135,150 +163,127 @@ class MultiSourceDocumentationBuilder {
+                }
+            }
+        } catch (error: any) {
+            console.warn(`Failed to read directory ${fullPath}:`, error.message);
```
fixed in [2feff61](https://github.com/SAP/open-ux-tools/pull/3680/commits/2feff613a468c85ff84f705493ea2780d919c30e)

devinea commented on packages/fiori-docs-embeddings/src/scripts/build-docs.ts:
```diff
@@ -135,150 +163,127 @@ class MultiSourceDocumentationBuilder {
+                            type: 'file',
+                            content
+                        });
+                    } catch (error: any) {
+                        console.warn(`Failed to read file ${fullEntryPath}:`, error.message);
```
fixed in [2feff61](https://github.com/SAP/open-ux-tools/pull/3680/commits/2feff613a468c85ff84f705493ea2780d919c30e)

devinea commented on packages/fiori-docs-embeddings/src/scripts/build-docs.ts:
```diff
@@ -135,150 +163,127 @@ class MultiSourceDocumentationBuilder {
-            }
+            return repoPath;
+        } catch (error: any) {
+            throw new Error(`Failed to clone/update repository ${repoName}: ${error.message}`);
```
fixed in [2feff61](https://github.com/SAP/open-ux-tools/pull/3680/commits/2feff613a468c85ff84f705493ea2780d919c30e)

devinea commented on packages/fiori-docs-embeddings/src/scripts/build-docs.ts:
```diff
@@ -4,8 +4,34 @@ import fetch from 'node-fetch';
 import { default as matter } from 'gray-matter';
 
+// Create promisified version of spawn for async/await
+const execCommand = (command: string, args: string[], options: any): Promise<{ stdout: string; stderr: string }> => {
```
fixed in [2feff61](https://github.com/SAP/open-ux-tools/pull/3680/commits/2feff613a468c85ff84f705493ea2780d919c30e)

devinea commented on packages/fiori-mcp-server/src/tools/services/text-embedding.ts:
```diff
@@ -0,0 +1,74 @@
+     * @returns True if initialized, false otherwise
+     */
+    isInitialized(): boolean {
+        return this.initialized && this.pipeline !== null;
```
fixed in [2feff61](https://github.com/SAP/open-ux-tools/pull/3680/commits/2feff613a468c85ff84f705493ea2780d919c30e)

pipe was dynamically downloaded original. But I removed this and added as dependency. 
So checking pipeline populated is not really needed.

heimwege reviewed (dismissed):
- Changeset ok
- Coverage is great
- Review comments addressed 
- Did not test manually

Yet there are some sonar findings

devinea commented:
> * Changeset ok
> * Coverage is great
> * Review comments addressed
> * Did not test manually
> 
> Yet there are some sonar findings

@heimwege sonar issues addressed (again). Thanks

vinayhospete reviewed (approved):
looks good to me
